archivist/dictmerge.py

```diff
@@ -28,3 +28,44 @@ def _dotdict(dct: Optional[dict]) -> Optional[dict]:
     if dct is None:
         return None
     return flatten(dct, reducer="dot")
+
+
+def assets_ext_attr(assets: list) -> list:
+    """Create list of assets with extended attribute(s)"""
+    extended_attributes_asset = [
+        asset
+        for asset in assets
+        if any(not item.startswith("arc_") for item in asset["attributes"])
+    ]
+    return extended_attributes_asset
+
+
+def attachment_identities_assets(assets_with_attachments: list) -> list:
+    """Create list of attachment identities"""
+    for asset in assets_with_attachments:
+        total_attachments_assets = {
+            item["arc_attachment_identity"]
+            for item in asset["attributes"]["arc_attachments"]
+        }
+    total_attachments_assets = list(filter(None, total_attachments_assets))
```

Comment:
this line appears to do nothing?

as a side issue never use the filter(), map() inbuilt functions - list comprehensions are faster

Reply:
if None entries are being rejected (is that the intention?) then

      total_attachments_assets = [a for a in total_attachments_assets if a is not None]

However if the get idiom is used as suggested above then this is not required.

Reply:
Yes, this line was from the old list comprehension I was using; it should have been removed when I updated to the correct comprehension. Apologies!